src/screens/ENSIntroSheet.tsx

```diff
@@ -65,33 +71,13 @@ export default function ENSIntroSheet() {
   const contentHeight = params?.contentHeight;
   const contentWidth = Math.min(deviceWidth - 72, 300);
 
-  const { ownedDomains, primaryDomain, nonPrimaryDomains } = useMemo(() => {
-    const ownedDomains = domains?.filter(
-      ({ owner }) => owner?.id?.toLowerCase() === accountAddress.toLowerCase()
-    );
-    return {
-      nonPrimaryDomains:
-        ownedDomains?.filter(({ name }) => accountENS !== name) || [],
-      ownedDomains,
-      primaryDomain: ownedDomains?.find(({ name }) => accountENS === name),
-    };
-  }, [accountAddress, accountENS, domains]);
-
-  const uniqueDomain = useMemo(() => {
-    return primaryDomain
-      ? primaryDomain
-      : nonPrimaryDomains?.length === 1
-      ? nonPrimaryDomains?.[0]
-      : null;
-  }, [nonPrimaryDomains, primaryDomain]);
-
-  useEffect(() => {
-    if (uniqueDomain?.name) {
-      prefetchENSAvatar(uniqueDomain.name);
-      prefetchENSCover(uniqueDomain.name);
-      prefetchENSRecords(uniqueDomain.name);
-    }
-  }, [uniqueDomain]);
+  const profileExists = useMemo(
+    () =>
+      Object.keys(ensRecords?.coinAddresses || {}).length > 1 ||
+      ensAvatar?.imageUrl ||
+      Object.keys(ensRecords?.records || {}).length > 0,
+    [ensAvatar?.imageUrl, ensRecords?.coinAddresses, ensRecords?.records]
+  );
```

Comment:
Probably doesn't need a memo. Isn't Object.keys pretty cheap operation?

Reply:
It seems to be linear https://stackoverflow.com/questions/7716812/object-keys-complexity
Not free, but perhaps not necessary to memoize unless it re-renders a ton